core/src/main/java/org/elasticsearch/indices/flush/IndicesSyncedFlushResult.java

```diff
@@ -41,7 +41,7 @@
 
     public IndicesSyncedFlushResult(Map<String, List<ShardsSyncedFlushResult>> shardsResultPerIndex) {
         this.shardsResultPerIndex = ImmutableMap.copyOf(shardsResultPerIndex);
-        this.shardCounts = calculateShardCounts(Iterables.concat(shardsResultPerIndex.values()));
+        this.shardCounts = calculateShardCounts(Iterables.flatten(shardsResultPerIndex.values()));
```

Comment:
@nik9000 Yes. Note that in the diff `com.google.common.collect.Iterables` was replaced by `org.elasticsearch.common.util.iterable.Iterables` in the imports.

The override of `com.google.commom.collect.Iterables#concat` that is being replaced here (by `org.elasticsearch.common.util.iterable.Iterables#flatten`) is a poorly named method.